packages/react-router-dom/modules/BrowserRouter.js

```diff
@@ -18,6 +18,12 @@ class BrowserRouter extends React.Component {
   history = createHistory(this.props)
 
   render() {
+    if (this.props.history) {
+      console.error(
+        '`<BrowserRouter history={...}` prop has been ignored. For custom history, ' +
+        'make sure to `import {Router}` and not `import {... as Router}`.')
+    }
+
```

Comment:
This is a minor nit, but I would prefer if there were spaces inside of the curly brackets. `import { Router }` and `import { ... as Router }`.